backend/src/schema/user.ts

```diff
@@ -32,49 +46,73 @@ const resolvers = {
 
       return `https://github.com/login/oauth/authorize?response_type=code&redirect_uri=${cbUrl}&client_id=${cliendID}`
-    } 
+    },
   },
   Mutation: {
-    authorizeWithGithub: async (_root: unknown, args: GitHubAuthCode):Promise<AuthResponse> => {
-      
-        if (!args.code) {
-          throw new UserInputError('GitHub code not provided')
-        }
+    authorizeWithGithub: async (
+      _root: unknown,
+      args: GitHubAuthCode
+    ): Promise<AuthResponse> => {
+      if (!args.code) {
+        throw new UserInputError('GitHub code not provided')
+      }
 
-        const { access_token } = await requestGithubToken(args.code)
-        
-        if (!access_token) {
-          throw new UserInputError('Invalid or expired GitHub code')
-        }
+      const { access_token } = await requestGithubToken(args.code)
+
+      if (!access_token) {
+        throw new UserInputError('Invalid or expired GitHub code')
+      }
+
+      let gitHubUser = await requestGithubUserAccount(access_token.toString())
+      // store gh token in user for now
+      gitHubUser = {
+        ...gitHubUser,
+        access_token: access_token.toString(),
+      }
+      if (!gitHubUser) {
+        throw new Error('No GitHub user found')
+      }
 
-        let gitHubUser = await requestGithubUserAccount(access_token.toString())
-        // store gh token in user for now
-        gitHubUser = {
-          ...gitHubUser,
-          access_token: access_token.toString()
-        }
-        if (!gitHubUser) {
-          throw new Error('No GitHub user found')
-        }
-        
-        const user = User.findOrCreateUserByGitHubUser(gitHubUser)
-        
-        const token = sign(
-          {
-            gitHubId: user.gitHubId,
-            gitHubToken: access_token.toString()
-          },
-          config.JWT_SECRET
-        )
+      const user = User.findOrCreateUserByGitHubUser(gitHubUser)
 
-        return {
-          user,
-          token
-        } 
+      const token = sign(
+        {
+          gitHubId: user.gitHubId,
+          gitHubToken: access_token.toString(),
+        },
+        config.JWT_SECRET
+      )
+
+      return {
+        user,
+        token,
+      }
     },
-    logout: (_root: unknown, _args:undefined, _context: AppContext):string => {
+    logout: (
+      _root: unknown,
+      _args: undefined,
+      _context: AppContext
+    ): string => {
       return 'logout'
     },
+    register: async (_root: unknown, args: LocalUser): Promise<string> => {
+      const username = args.username
+      const email = args.email
+      const password = args.password
+
+      try {
+        await pool.query('BEGIN')
+        const queryText =
+          'INSERT INTO users(username, email, password) VALUES($1, $2, $3)'
+        await pool.query(queryText, [username, email, password])
+        await pool.query('COMMIT')
+        return 'Registered'
+      } catch (error) {
+        await pool.query('ROLLBACK')
+      }
+
```

Comment:
I would put the query into model/user.ts file as a single function "registerNewUser" or something like that. Also is `await pool.query('BEGIN')` `await pool.query('COMMIT')` necessary? Query could also return the newly created user in response

Reply:
Does `return 'Register failed'` return success status code? I think throwing error would be better if the registering fails